galsim/randwalk.py

```diff
@@ -100,21 +117,57 @@ class RandomWalk(GSObject):
     _is_analytic_x = False
     _is_analytic_k = True
 
-    def __init__(self, npoints, half_light_radius, flux=1.0, rng=None, gsparams=None):
+    def __init__(self,
+                 npoints,
+                 half_light_radius=None,
+                 flux=1.0,
+                 profile=None,
+                 rng=None,
+                 gsparams=None):
+
         from .random import BaseDeviate
 
         self._npoints = int(npoints)
-        self._half_light_radius = float(half_light_radius)
-        self._flux = float(flux)
-        self._gsparams = GSParams.check(gsparams)
 
-        # we will verify this in the _verify() method
         if rng is None:
             rng = BaseDeviate()
         self._rng=rng
+
+        self._flux = float(flux)
         self._verify()
 
-        self._set_gaussian_rng()
+        if profile is None:
+            if half_light_radius is None:
+                # use runtime error, because this is a fatal error
+                raise RuntimeError("send a half_light_radius "
```

Comment:
I think this would normally be a TypeError.  (What you usually get when you don't provide required arguments to a function.)  Or, I've usually been calling this kind of thing a `GalSimIncompatibleValuesError`, which is a subtype of TypeError.

Reply:
fixed